tests/Unit/Framework/MockDistributedObject.hpp

```diff
@@ -129,6 +138,54 @@ struct LocalCoreId {
   size_t value;
 };
 
+inline bool operator==(const LocalCoreId& lhs,
+                       const LocalCoreId& rhs) noexcept {
+  return lhs.value == rhs.value;
+}
+
+inline bool operator!=(const LocalCoreId& lhs,
+                       const LocalCoreId& rhs) noexcept {
+  return not(lhs==rhs);
+}
+
+/// Wraps a size_t representing the global core number.
+struct GlobalCoreId {
+  size_t value;
+};
+
+inline bool operator==(const GlobalCoreId& lhs,
+                       const GlobalCoreId& rhs) noexcept {
+  return lhs.value == rhs.value;
+}
+
+inline bool operator!=(const GlobalCoreId& lhs,
```

Comment:
Why inline? I never know when to mark functions inline actually..

Reply:
I'm not sure when they should be inlined either.  I did it here because other examples I looked at were inlined.

Reply:
Roughly, `inline` allows the definition of a non-template function in a header file without the linker complaining.  Functions defined in a class body are implicitly inline.